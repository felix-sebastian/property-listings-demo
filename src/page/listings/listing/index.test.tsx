import React from "react";
import { render } from "@testing-library/react";
import Listing from "./";

test("uses fallbacks when the id is invalid", () => {
  const result = render(
    <Listing
      price={0}
      status={""}
      image=""
      street="123 Example St"
      suburb="Melbourne"
      state="VIC"
      postcode="3000"
    />
  );

  expect(result.getByTestId("image")).toHaveAttribute(
    "alt",
    "123 Example St Melbourne"
  );

  expect(result).toMatchSnapshot();
});
