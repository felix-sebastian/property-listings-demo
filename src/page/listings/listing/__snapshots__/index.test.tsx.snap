// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`uses fallbacks when the id is invalid 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-AxhUy cWLZif"
      >
        <div
          class="sc-AxiKw cGtQMs"
        >
          <div
            data-testid="price"
          >
            $0
          </div>
           
          <div
            class="sc-AxjAm iMxqKn"
          >
            <span
              data-testid="text"
            />
            <div
              class="sc-AxirZ knwJVO"
              color="grey"
              data-testid="dot"
            />
          </div>
        </div>
        <img
          alt="123 Example St Melbourne"
          class="sc-AxgMl htDxvp"
          data-testid="image"
          src=""
        />
        <div
          class="sc-AxhCb cinyOp"
        >
          123 Example St
          <br />
          Melbourne
          <br />
          VIC
           
          3000
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AxhUy cWLZif"
    >
      <div
        class="sc-AxiKw cGtQMs"
      >
        <div
          data-testid="price"
        >
          $0
        </div>
         
        <div
          class="sc-AxjAm iMxqKn"
        >
          <span
            data-testid="text"
          />
          <div
            class="sc-AxirZ knwJVO"
            color="grey"
            data-testid="dot"
          />
        </div>
      </div>
      <img
        alt="123 Example St Melbourne"
        class="sc-AxgMl htDxvp"
        data-testid="image"
        src=""
      />
      <div
        class="sc-AxhCb cinyOp"
      >
        123 Example St
        <br />
        Melbourne
        <br />
        VIC
         
        3000
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
